### ISSUE TITLE
Suggestion: scrap grouping

### ISSUE THREAD

**Issue by Jonteiro** [bozzobrain/LethalCompanyShipMaid#13]
Hi, i usually group my scrap into a one-handed pile, and a two-handed pile, is there a way to add a config option to group scrap by type instead of having to manually add each scrap item to the override list?

**Comment by bozzobrain** [comment]
I'm assuming that I understand your request. 
1. This feature should be added already. 
  - Try disabling the 'UseItemTypePlacementOverrides'
  - Enable both 'UseOneHandedPlacementOverrides' and 'UseTwoHandedPlacementOverrides'
2. You should not have to 'manually' add items to the override list. 
  - Depends on what you mean by manually
  - Using the keybind 'J' or what ever is set by 'SetObjectTypePositionKey' you can set the position for not only individual items (assuming you have 'UseItemTypePlacementOverrides' enabled) or set the position for one and two-handed objects if those overrides are enabled. Doing this keybind in game will write to the config file the location of one-handed/two-handed or object type positions.
3. If this solves your issue by me explaining these options, let me know and then I feel need to make this more clear in the readme which I feel may be lacking and confusing and outdated.
4. Thanks for reaching out and let me know if this solves your suggestion.

**Comment by Jonteiro** [comment]
Hi, I should've made my issue clearer, apologies, what i meant was, that i had tried one handed scrap override and two handed but that also includes the rest of one and two handed items on the ship, i usually dont use the cupboard to store items since with no mods they fall on load and they become a hassle to get due to camera position and angles, that said, i organize my ship, in the following way, 1 pile of any one handed scrap, 1 pile of two handed scrap, and then several piles for the different tools (flashlights, shovels, shotguns, jetpacks, ladders and more). What i wanted was a way to keep the rest of the items position saved but then take ANY SCRAP (not any item) and put it in its according pile without overriding one handed items saved position. Hopefully this is clearer, thanks :)

**Comment by bozzobrain** [comment]
Okay, re-iterating in my own words to make sure we are communicating clearly.

- You want to have the One Handed and Two handed position overrides Enabled to organize these items to their own piles. (Configuration [UseOneHandedPlacementOverrides] and [UseTwoHandedPlacementOverrides])
- You want to have excluded items that don't get sorted to those piles.  (Similar to Configuration [SortingDisabledList])
- BUT you want to have the excluded objects (Shovels/Shotguns/etc.) to have specific organization locations.  (No Current Solution)

If those statements are correct, I totally see your intent and why you would want this. There may be a relatively easy way to accomplish this that I can play with.

It sounds like a mix of UseOneHandedPlacementOverrides/UseTwoHandedPlacementOverrides and UseItemTypePlacementOverrides. 

My goal would be to incorporate this but try to keep it from being confusing, which may be the hard part, but if that all sounds good, I can work towards trying to implement.

**Comment by Jonteiro** [comment]
Yes that is all correct but my suggestion to this problem would be a config bool to group scrap by their handed type when using item type override (Type Scrap ?) instead of grouping everything and mixing the one handed and two handed overrides with item type override

**Comment by bozzobrain** [comment]
I'm away from my PC with the game installed, but I think I see what your proposing. 

To be sure that I can replicate this later, can you send me a copy of your config file?

Maybe I can throw together a release candidate and have you try it out before releasing it as a full version.

Edit: Also, can you try to have all three overrides enabled and see if that allows you to do what you are requesting?

**Comment by Jonteiro** [comment]
yes I'd love to test for your mod :) thank you for listening to my suggestions, i will try all 3 overrides but i will only be home in about 4 hours so I'll post another reply then

**Comment by Jonteiro** [comment]
After testing with all three overrides it seems to work, but the weird part is that after wiping the saved values from the config they somehow persisted (it was the same save used before) so the old items had their coords saved but after running into the facility and getting new scrap types (scrap that was not saved before by type) it seems to work as intended above! and old items kept their placement with new scrap items being thrown on the old pile! Success :D

**Comment by bozzobrain** [comment]
Nice! I'm glad to hear that that option works. It just so happens that the order that I check the configuration options was the ItemTypeOverride first, so objects are caught there and not processed as One-Handed or Two-Handed. That means that as long as an object has an Item Type override location it will be caught by ItemType and not by One or Two handed.

I'm interested in the recovering configuration after wiping it. I did make an effort to capture item locations on game load, since before I was not storing the locations in the config file. It seems that that old setup is still logging object locations on boot, so some things could have been recovered. Although if they do get stored for organizing, it shouldn't re-write the config file for those objects, so do look out for that.

Glad this worked! Closing the issue.

Thanks for the banter and your help looking for ways to improve!
